cmake/compiler/xcc/compiler_flags.cmake

```diff
@@ -6,6 +6,10 @@ if(CC STREQUAL "clang")
   endif()
 
   include(${ZEPHYR_BASE}/cmake/compiler/clang/compiler_flags.cmake)
+
+  if($ENV{XCC_NO_G_FLAG})
+    unset(GCC_NO_G_FLAG 1)
+  endif()
```

Comment:
Sorry, this was not what I meant with this comment: https://github.com/zephyrproject-rtos/zephyr/pull/42017#discussion_r795544812

what I meant was that you should do this instead:
```
  include(${ZEPHYR_BASE}/cmake/compiler/clang/compiler_flags.cmake)

  # Now, let's overwrite the flags that are different in xcc/clang.
  if($ENV{XCC_NO_G_FLAG})
    set_compiler_property(PROPERTY debug)
  endif()
```

and then revert this commit:
https://github.com/zephyrproject-rtos/zephyr/commit/deeb98da5328c6fbdfc4ad1c7638a205d86f83e7

Reply:
Updated, and reverted the old commit.